proto/auth/auth.proto

```diff
@@ -18,9 +19,14 @@ message AuthContainer {
   AuthBody protectedPayload = 1;      // envelope body, a marshalled protobuf data or it can be null
   // if the length of senderCertHash received is not N bytes, as described in hashAlgorithm, then the protobuf
   // message either is not AuthContainer type, or is corrupted. Otherwise, the
-  // receiver may not have the sender's signing certificate
+  // receiver may not have the sender's signing certificate.
+  // If `cipherData` is defined and valid, then `protectedPayload` is null and actual payload is
+  // defined in the `cipherData.cipherData` member. Once the `CipherBlock` is decrypted,
+  // `protectedPayload` is defined by the auth stack.
   org.lfedge.eve.common.HashAlgorithm algo = 2;        // hash algorithm used by sender Cert
   bytes   senderCertHash = 3;    // N bytes in length, 1st N bytes of sender siging cert sha256 hash
   bytes    signatureHash = 4;    // signature of the sha256 hash of the payload
   bytes    senderCert = 5;       // full senderCert needed for some payloads
+  org.lfedge.eve.common.CipherContext cipherContext = 6; // context for decrypting payload if defined
+  org.lfedge.eve.common.CipherBlock cipherData = 7; // encrypted payload if defined
```

Comment:
For current encryption we have the CipherContext and Cipherblock in different parts of the API and the CipherBlock references the CipherContext using the cipherContextId. Do we expact to do that here, or will the contextId be empty and the reference is implicit by the contect and block being in the same AuthContainer?

Reply:
ContextID should be valid for CipherBlock and CipherContext, in the decryption routine I check that those two should match. So there is no any difference here.

Reply:
OK